prog/dftb+/lib_solvation/solvinput.F90

```diff
@@ -0,0 +1,82 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2020  DFTB+ developers group                                               !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+#:include 'common.fypp'
+
+!> Helper routines to handle input to solvation models
+module dftbp_solvinput
+  use dftbp_accuracy, only : dp
+  use dftbp_born, only : TGeneralizedBorn, TGBInput, init
+  use dftbp_solvation, only : TSolvation
+  implicit none
+  private
+
+  public :: TSolvationInp, createSolvationModel, writeSolvationInfo
+
+  !> Collection of solvation input data
+  type :: TSolvationInp
+
+    !> Generalized Born input data
+    type(TGBInput), allocatable :: GBInp
+
+  end type TSolvationInp
+
+  !> Wrapper to create a solvation model from its respective input
+  interface createSolvationModel
+    module procedure :: createGeneralizedBornModel
+  end interface
+
+contains
+
+  !> Print the solvation model used
```

Comment:
For user experience, if they set a named solvent it would be useful to print out the name in the output, i.e. store a representation of the initialization of the model somewhere. Possibly the stdOut stream should have more information at the initialization stage (temperature, shift, ...). Of course they should be referring to the dftb_pin for this, but it doesn't always happen.

Reply:
Does DFTB+ print the name of the SK-files used? The verbosity is at the same as with the dispersion corrections, maybe I can come up with an option to make it more verbose and print a detailed summary on the solvation model.
This will be definitely necessary for the final GBOBC+SASA+CM5+HB model we are aiming for.